src/Analyzers/CSharp/Tests/UseUTF8StringLiteral/UseUTF8StringLiteralTests.cs

```diff
@@ -0,0 +1,596 @@
+                TestCode =
+@"
+public class C
+{
+    public void M()
+    {
+        var x = new byte[] { 65, 66, 67 };
+    }
+}",
+                LanguageVersion = LanguageVersion.CSharp10
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseUTF8StringLiteral)]
+        public async Task TestNotWithoutInitializer()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode =
+@"
+public class C
+{
+    public void M()
+    {
+        var x = new byte[10];
+    }
+}",
+                LanguageVersion = LanguageVersion.Preview
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseUTF8StringLiteral)]
+        public async Task TestNotInExpressionTree()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode =
+@"
+using System;
+using System.Linq.Expressions;
+
+public class C
+{
+    public void M()
+    {
+        N(() => new byte[] { 65, 66, 67 });
+    }
+
+    public void N(Expression<Func<byte[]>> f)
+    {
+    }
+}",
+                LanguageVersion = LanguageVersion.Preview
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseUTF8StringLiteral)]
+        public async Task TestNotWhenNotByteArray()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode =
+@"
+public class C
+{
+    public void M()
+    {
+        var x = new int[] { 65, 66, 67 };
+    }
+}",
+                LanguageVersion = LanguageVersion.Preview
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseUTF8StringLiteral)]
+        public async Task TestNotWhenOptionNotSet()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode =
+@"
+public class C
+{
+    public void M()
+    {
+        var x = new byte[] { 65, 66, 67 };
+    }
+}",
+                EditorConfig = @"
+[*.cs]
+csharp_style_prefer_utf8_string_literal = false
+",
+                LanguageVersion = LanguageVersion.Preview
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseUTF8StringLiteral)]
+        public async Task TestNotWhenNonLiteralElement()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode =
+@"
+public class C
+{
+    public void M()
+    {
+        var x = new byte[] { 65, GetB(), 67 };
+    }
+
+    public byte GetB() => 66;
+}",
+                LanguageVersion = LanguageVersion.Preview
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseUTF8StringLiteral)]
+        public async Task TestSimpleByteArray()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode =
+@"
+public class C
+{
+    public void M()
+    {
+        var x = {|IDE0230:new byte[] { 65, 66, 67 }|};
```

Comment:
This is more of a stylistic opinion. I usually find the shorter markup syntax easier to read. But it doesn't matter a lot.

```suggestion
        var x = [|new byte[] { 65, 66, 67 }|];
```

Reply:
Thanks for the tip, didn't know.